SonarScanner does not work

While working on #78 I have discovered that SonarScanner does not work any more:

```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.157 s
[INFO] Finished at: 2019-08-18T06:39:15Z
[INFO] Final Memory: 17M/194M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.2:sonar (default-cli) on project JTango-root: Unable to execute SonarQube: Fail to download libraries from server -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :JTango-root
The command "mvn sonar:sonar -Ptravis" exited with 1.
```

[Travis build#364.3](https://travis-ci.org/tango-controls/JTango/jobs/573334591)

Following [this](https://community.sonarsource.com/t/fail-to-download-libraries-from-server/8968/2) I have upgraded sonar-maven-plugin to the latest `3.6.0.1398` version 

Which is not compatible with current configuration:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 31.696 s
[INFO] Finished at: 2019-08-18T06:47:32Z
[INFO] Final Memory: 33M/364M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar (default-cli) on project JTango-root: Validation of project reactor failed:
[ERROR]   o The property "sonar.branch.name" must not be used together with the deprecated "sonar.branch"
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :JTango-root
The command "mvn sonar:sonar -Ptravis" exited with 1.
```

[Travis build #366.3](https://travis-ci.org/tango-controls/JTango/jobs/573335813)

After removing deprecated `sonar.branch` property build still fails:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 28.129 s
[INFO] Finished at: 2019-08-18T06:56:44Z
[INFO] Final Memory: 33M/366M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar (default-cli) on project JTango-root: Project was never analyzed. A regular analysis is required before a branch analysis -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :JTango-root
The command "mvn sonar:sonar -Ptravis" exited with 1.
```

[Travis build#368.3](https://travis-ci.org/tango-controls/JTango/jobs/573337108)

We need to invest some time into proper SonarScanner configuration...

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,6 +4,9 @@ sudo: false
 services:
 - docker
 
+git:
+  depth: false
+
 env:
   - TANGO_HOST=localhost:10000
 

--- a/README.md
+++ b/README.md
@@ -10,19 +10,20 @@
 
 ## SonarCloud
 
-[![Quality gate](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=alert_status)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
+[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=alert_status)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
 
-[![Maintainability](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=sqale_rating)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
-[![Reliability](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=reliability_rating)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
-[![Security](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=security_rating)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
+[![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=sqale_rating)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
+[![Reliability Rating](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=reliability_rating)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
+[![Security Rating](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=security_rating)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
 
-[![Bugs](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=bugs)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
-[![Code smells](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=code_smells)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
-[![Vulnerabilities](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=vulnerabilities)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
-[![Tech debt](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=sqale_index)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
+[![Bugs](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=bugs)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
+[![Code Smells](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=code_smells)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
+[![Vulnerabilities](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=vulnerabilities)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
+[![Technical Debt](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=sqale_index)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
 
 
-[![Lines of code](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root%3Ajtango-9-lts&metric=ncloc)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root%3Ajtango-9-lts)
+[![Lines of Code](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=ncloc)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
+[![Duplicated Lines (%)](https://sonarcloud.io/api/project_badges/measure?project=org.tango-controls%3AJTango-root&metric=duplicated_lines_density)](https://sonarcloud.io/dashboard?id=org.tango-controls%3AJTango-root)
 
 # JavaDoc
 

